### PR TITLE
ci: run the enforcement gate corpus as its own check, so the red direction is visible

### PR DESCRIPTION
**Pull request by Oleg67**
**What.** A `gates` CI job and a `make test-gates` target that run the two enforcement test files on their own. No new tests, no behaviour change, no new dependencies.

**Why.** The pipeline currently exercises these gates only in their passing direction. `ci.yml` has ten jobs — `test`, `coverage`, `sonarqube`, `pylint`, `vulture`, `versions`, `spec-coverage`, `validate`, `validate-kits` — and none of them asserts that a seeded violation is *rejected*. So a green run says the checks ran, not that they can fail.

That is this repository's own argument turned on its CI. #89 existed because a green exit did not mean the spec was implemented; the same reasoning applies to a green pipeline that never demonstrates a gate failing.

**The assertions already exist** — this is about visibility, not coverage. `tests/test_enforcement_empty_codebase.py` and `tests/test_enforcement_empty_scan.py` drive the real CLI and assert exit codes in both directions:

| Fixture | Expected |
|---|---|
| Registered codebase, one **unmarked** source file | exit 2, `code-no-marker` |
| Same claim, **zero** code files | exit 2 — scan size does not change the verdict |
| Correctly **marked** implementation | exit 0 |
| Spec-first project, nothing claimed, no code | exit 0 — nothing claimed, nothing owed |

Those run inside a suite of ~4,850 tests, so today the red direction is provable only by knowing the tests exist and going to read them. After this, the pipeline shows it as *"Enforcement Gates (seeded violations must fail)"*.

**Verified load-bearing rather than decorative.** Restoring the early return that cross-validation used to have — `if not results.parsed_code_files_full:` — makes **5 of the 39** fail. So the check catches the regression it exists to catch. Green on `main` as it stands.

**Scope.** Deliberately limited to the enforcement corpus. The semantic-coverage tests are advisory and currently have an interpreter-dependent failure of their own (#120), so including them would make this check red for a reason unrelated to enforcement — which would defeat the point of a named gate check.

**Non-goals.** Not making any check *required* — that is branch protection and not something a commit can do. Not adding a CLI-level negative fixture repo; the existing tests already drive the real CLI end-to-end, and a second mechanism would duplicate them.

**Gates.** The corpus itself: 39 passed. Repository unaffected, as expected for a Makefile/workflow-only change — `cfs validate` 0 errors 0 warnings, `spec-coverage --system studio` granularity 0.4612 with all thresholds met, `pylint` clean. Workflow YAML parses with the new job registered.

## Summary by CodeRabbit

* **Tests**
  * Added a dedicated enforcement gate check for validating known-good and known-bad scenarios.
  * Added a separate CI pipeline check to report enforcement results independently from the main test suite.

* **Documentation**
  * Updated available development commands to include the new enforcement gate check.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -34,6 +34,29 @@ jobs:
...
+  # that a gate can fail -- and a gate that never fails is worthless. This job
+  # makes the red direction visible in the pipeline: known-bad fixtures must be
+  # rejected, known-good fixtures must stay clean.
+  gates:
+    name: Enforcement Gates (seeded violations must fail)
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```
[the remark this answers is not in the document]
Comment (Oleg67) [reply .github/workflows/ci.yml:52]: Fixed for the job this PR adds, in `6eac6099`: `persist-credentials: false` on the checkout and a job-level `permissions: contents: read`.

I scoped it to the `gates` job rather than sweeping all ten checkout steps. Two reasons:

1. The exposure you describe is the pre-existing posture of this workflow, not something this PR introduces — `test`, `coverage`, `pylint`, `vulture`, `spec-coverage`, `validate` and `validate-kits` all already run repository code with the same persisted-credentials default. Hardening one of ten steps in a CI-visibility PR while quietly changing behaviour for the nine I did not touch seemed like the wrong trade; a workflow-level `permissions` block in particular would alter the token scope for every existing job in a PR whose stated purpose is adding a check.
2. The job-level shape matches what `code-ranker.yml` already does in this repo (least privilege declared per job, with a comment for what is deliberately omitted), so it fits house style.

Happy to do the full sweep — all ten checkout steps plus a workflow-level default — as a separate PR if @ainetx wants it; it is a one-line-per-step change and I would rather it land as its own reviewable unit.

One correction to the finding as written: on fork pull requests, which is how this branch runs, the token is read-only against a public repository, so the practical value of exfiltrating it is close to nil. The change is still worth making as defence in depth for the `push`-to-`main` trigger, where the token is not read-only — I just would not want the severity read as an active leak.

Verified: the workflow still parses, and the two files `make test-gates` runs use no real git and no network (their only git reference is an empty `.git` directory created inside a temp fixture), so removing the credential has no effect on them. CI on the new head is the confirming check.

**Review by ainetx — APPROVED** [review]
